src/main/java/com/example/music/backend/common/security/WebSecurityConfig.java

```diff
@@ -0,0 +1,122 @@
+import com.example.music.backend.common.security.googleLogin.CustomOAuth2UserService;
+import com.example.music.backend.common.security.handler.CustomAccessDeniedHandler;
+import com.example.music.backend.user.domain.Role;
+import com.example.music.backend.user.service.UserService;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.web.access.AccessDeniedHandler;
+import org.springframework.security.web.authentication.AuthenticationSuccessHandler;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@PropertySource("classpath:constants.properties")
+@Configuration
+@EnableWebSecurity
+public class WebSecurityConfig extends WebSecurityConfigurerAdapter {
+    private static final String COOKIES = "JSESSIONID";
+    private final CustomOAuth2UserService oauthUserService;
+    private final UserService userService;
+
+    @Value("${maximum.session.count}")
+    private int MAXIMUM_SESSIONS_COUNT;
+
+    private final UserDetailsService userDetailsService;
+    private final PasswordEncoder encoder;
+
+    public WebSecurityConfig(CustomOAuth2UserService oauthUserService, UserService userService, UserDetailsService userDetailsService, PasswordEncoder encoder) {
+        this.oauthUserService = oauthUserService;
+        this.userService = userService;
+        this.userDetailsService = userDetailsService;
+        this.encoder = encoder;
+    }
+
+    @Bean
+    public AccessDeniedHandler accessDeniedHandler() {
+        return new CustomAccessDeniedHandler();
+    }
+
+    @Bean
+    RestAuthenticationEntryPoint authenticationEntryPoint() {
+        return new RestAuthenticationEntryPoint();
+    }
+
+    @Override
+    protected void configure(final AuthenticationManagerBuilder authenticationManagerBuilder) throws Exception {
+        authenticationManagerBuilder.userDetailsService(userDetailsService).passwordEncoder(encoder);
+    }
+
+    @Override
+    protected void configure(final HttpSecurity http) throws Exception {
+        http
+                .exceptionHandling().accessDeniedHandler(accessDeniedHandler())
+
+                .and()
+                .httpBasic()
```

Comment:
why do you need httpBasic?

Reply:
it needs to be processed authenticationEntryPoint